modules/imgproc/test/test_color.cpp

```diff
@@ -3072,20 +3072,40 @@ TEST(ImgProc_RGB2YUV, regression_13668)
     EXPECT_EQ(res, ref);
 }
 
-TEST(ImgProc_cvtColorTwoPlane, missing_check_17036)  // test can be removed if required feature is implemented
+TEST(ImgProc_cvtColorTwoPlane, regression_17036)
 {
-    std::vector<uchar> y_data(700 * 480);
-    std::vector<uchar> uv_data(640 * 240);
+    RNG rng;
 
-    Mat y_plane_padding(480, 640, CV_8UC1, y_data.data(), 700);  // with stride
-    Mat uv_plane(240, 320, CV_8UC2, uv_data.data());
+    std::vector<uchar> y(640 * 480,  (uchar)rng.uniform(16, 235));
+    Mat y_mat(480, 640, CV_8UC1, y.data());
 
-    Mat result;
+    std::vector<uchar> uv(640 * 240, (uchar)rng.uniform(16, 240));
+    Mat uv_mat(240, 320, CV_8UC2, uv.data());
+
+    std::vector<uchar> uv_strided(700 * 240, 0);
+    for (size_t i = 0; i < 240; ++i) {
+        for (size_t j = 0; j < 640; j++) {
+            uv_strided[i * 700 + j] = uv[i * 640 + j];
```

Comment:
`cv::Mat::at<>` takes stride in account. No need to create vector, you can just copy Mat elements. As alternative you can try `cv::Mat::copyTo`, but I'm not sure if it supports output with step.

Reply:
Fixed by switching to the former option.